code/datums/components/space_allergy.dm

```diff
@@ -4,6 +4,12 @@
 		return COMPONENT_INCOMPATIBLE
 
 	RegisterSignal(parent, COMSIG_ENTER_AREA, PROC_REF(entered_area))
+	entered_area(parent, get_area(parent))
+
+/datum/component/planet_allergy/Destroy(force)
+	var/mob/living/as_living = parent
+	as_living.remove_status_effect(/datum/status_effect/planet_allergy)
+	return ..()
```

Comment:
Feel like we should just keep a reference to the status effect we apply and `qdel_null` it on destroy